backend/replication/db_test.go

```diff
@@ -1,68 +1,97 @@
 package replication
 
 import (
-	"reflect"
+	"context"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
+
 	"github.com/upfluence/sql"
 	"github.com/upfluence/sql/backend/static"
 	"github.com/upfluence/sql/sqlparser"
 )
 
+type mockDB struct {
+	static.DB
+
+	Called bool
+}
+
+func (mdb *mockDB) Query(_ context.Context, q string, vs ...interface{}) (sql.Cursor, error) {
+	mdb.Called = true
+
+	return nil, nil
+}
+
 type mockParser map[string]sqlparser.StmtType
 
 func (p mockParser) GetStatementType(q string) sqlparser.StmtType {
 	return p[q]
 }
 
 func TestPickDB(t *testing.T) {
-	var db0, db1 static.DB
-
 	tests := []struct {
-		name  string
-		query string
-		args  []interface{}
-		db    *db
-		want  sql.DB
+		name   string
+		query  string
+		args   []interface{}
+		parser sqlparser.SQLParser
```

Comment:
instead of passing a `parser` and `query` just pass a `sqlparser.StmtType`. and hard code the statement in the body of the test

Reply:
@tgallice you forgot  the `query` attr, it does not make sense to leave it in the test case. It has to be equal to the key of the mock parser.